File: lib/layer2_core/dark/e_on_tertiary_container_dark.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/dark/system.dart';
import '../../layer1_raw/dark/standard.dart';
import '../../layer1_raw/dark/blue.dart';
import '../../layer1_raw/dark/green.dart';
import '../../layer1_raw/dark/red.dart';
import '../../layer1_raw/dark/nothing.dart';

class EOnTertiaryContainerDark {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyd36;
      case 'standard': return cstd36;
      case 'blue': return cbd36;
      case 'green': return cgd36;
      case 'red': return crd36;
      case 'nothing': return cnod36;
      default: return cstd36;
    }
  }
}

File: android/src/main/kotlin/com/example/design_engine/layer2_core/dark/EOnTertiaryContainerDark.kt
package com.example.design_engine.layer2_core.dark
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EOnTertiaryContainerDark {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyd36
            AppTheme.STANDARD -> R.color.cstd36
            AppTheme.BLUE -> R.color.cbd36
            AppTheme.GREEN -> R.color.cgd36
            AppTheme.RED -> R.color.crd36
            AppTheme.NOTHING -> R.color.cnod36
        }
    }
}
